docs/DEV.md

```diff
@@ -0,0 +1,161 @@
+
+* Open the Browser Console using Firefox's top menu at `Tools > Web Developer > Browser Console`. This will display Shield (loading/telemetry) and log output from the add-on.
+
+See [TESTPLAN.md](./TESTPLAN.md) for more details on how to see this add-on in action and hot it is expected to behave.
+
+## Automated launch of Firefox with add-on installed
+
+`$ npm run firefox` starts Firefox and automatically installs the add-on in a new profile and opens the browser console automatically.
+
+Note: This runs in a recently created profile, and the study variation/branch is overridden by a preference in the FIREFOX_PREFERENCES section of `test/utils.js`.
+
+## Automated testing
+
+`npm run test` verifies the telemetry payload as expected at Firefox startup and add-on installation in a clean profile, then does **optimistic testing** of the _commonest path_ though the study for a user
+
+* prove the notification bar ui opens
+* _clicking on the left-most button presented_.
+* verifying that sent Telemetry is correct.
+
+Code at [/test/functional_test.js](/test/functional_test.js).
+
+Note: The functional tests are using async/await, so make sure you are running Node 7.6+
+
+The functional testing set-up is imported from [https://github.com/mozilla/share-button-study]() which contains plenty of examples of functional tests relevant to Shield study addons.
+
+## Watch
+
+You can automatically build recent changes and package them into a `.xpi` by running the following from the top level directory:
+
+`$ npm run watch`
+
+Now, anytime a file is changed and saved, node will repackage the add-on. You must reload the add-on as before, or by clicking the "Reload" under the add-on in _about:debugging_. Note that a hard re-load is recommended to clear local storage. To do this, simply remove the add-on and reload as before.
+
+Note: This is currently only useful if you load the extension manually - it has no effect when running `npm run firefox`.
+
+## Directory Structure and Files
+
+```
+├── .circleci             # setup for .circle ci integration
+│   └── config.yml
+├── .eslintignore
+├── .eslintrc.js          # mozilla, json
+├── .gitignore
+├── LICENSE
+├── README.md             # (this file)
+├── addon                 # Files that will go into the addon
+│   ├── Config.jsm        # Study-specific configuration regarding branches, eligibility etc
+│   ├── StudyUtils.jsm    # (copied in during `prebuild`)
+│   ├── bootstrap.js      # LEGACY Bootstrap.js
+│   ├── chrome.manifest   # (derived from templates)
+│   ├── icon.png
+│   ├── install.rdf       # (derived from templates)
+│   ├── lib               # JSM (Firefox modules)
+│   │   └── Feature.jsm   # contains study-specific privileged code
+│   └── webextension      # study-specific embedded webextension
+│       ├── .eslintrc.json
+│       ├── background.js
+│       ├── icons
+│       │   ├── Anonymous-Lizard.svg
+│       │   ├── DogHazard1.svg
+│       │   ├── Grooming-Cat-Line-Art.svg
```

Comment:
We probably want to remove these three from this [commit](https://github.com/mozilla/shield-studies-addon-template/pull/49/commits/ec44d6d53181735a715270a514c97f12a3094162) in the PR:
* Anonymous-Lizard.svg
* DogHazard1.svg
* Grooming-Cat-Line-Art.svg